### PR TITLE
 Use l10n.js instead of a custom l10n code. #77 

### PR DESCRIPTION
**Pull request by Timmaks**
port to l10n.js.

### CODE REVIEW

**Comment by willyaranda** [comment]
Awesome! I will give it a shot this week.

**Comment by Timmaks** [comment]
https://www.transifex.com/projects/p/openwapp/ is ready to go!

**Comment by Timmaks** [comment]
@willyaranda how is going with you review?

**Comment by willyaranda** [comment]
I am returning from vacation without computer. I can review from this
Wednesday!

**Comment by Timmaks** [comment]
no worries, it is just that transifex is getting updated and i don't want those people waiting to long to see there work in the product

hope you had a good vacation :-|)

File: app/scripts/localisations/translation.js [hunk 1]
```diff
@@ -1168,197 +1168,7 @@ define([], function () {
       simCard: 'SIM-Karte'
```
Comment (willyaranda) [inline app/scripts/localisations/translation.js]: Please, could you remove all translations from this file?

File: app/scripts/localisations/translation.js [hunk 2]
```diff
@@ -1168,197 +1168,7 @@ define([], function () {
...
-      accountExpiration: 'Expiratie datum',
-      upgradeNow: 'Nu upgraden',
-      notSquarePicture: 'De geselecteerde afbeelding is niet vierkant. Er ' +
-      'zullen dan strepen te zien zijn. Wilt u dit voorkomen, ga dan naar ' +
-      'Gallerij, en crop/edit de foto, en zorg voor een 1:1 verhouding.',
-      simRequired: 'Er is een SIM-kaart benodigd. Zonder een SIM-kaart werkt ' +
-      'de app waarschijnlijk niet goed.',
-      multiSIMdetectedWarn: 'Er zijn meerder SIM-kaarten aanwezig. ' +
-      'OpenWapp zal de eerste SIM-kaart gebruiken.'
+      all: 'Alle'
```
Comment (willyaranda) [inline app/scripts/localisations/translation.js]: Why this? :)

File: app/scripts/views/login.js [hunk 3]
```diff
@@ -72,23 +72,20 @@ define([
...
-          stringId = 'countryNotDetectedOnLogin';
-          message = l10n[stringId];
+          message = navigator.mozL10n.get('countryNotDetectedOnLogin');
         }
         else if (this.possibleSimCards.length === 1) {
           this.selectedSimCard = this.possibleSimCards[0];
           var interpolate = global.l10nUtils.interpolate;
-          stringId = 'countryDetectedOnLogin';
-          message = interpolate(l10n[stringId], {
+          stringId = navigator.mozL10n.get('countryDetectedOnLogin');
```
Comment (willyaranda) [inline app/scripts/views/login.js]: this is not an `Id`, its a string, so change the naming to `message`

File: app/index.html [hunk 4]
```diff
@@ -7,9 +7,13 @@
   <link rel="stylesheet" href="emoji/emoji-20px.css">
   <link rel="stylesheet" href="emoji/emoji-45px.css">
   <link rel="stylesheet" href="styles/main.css">
-  <!-- Once we change our actual l10n framework to l10n.js (or better, l20n.js),
-       we need to merge fxosrate strings to our new l10n files -->
-  <link rel="resource" type="application/l10n" href="components/fxosRate/locales/locales.ini" />
```
Comment (willyaranda) [inline app/index.html]: You are removing the `fxosrate` files, but you have not added the strings to the new `properties` file, right?

File: app/scripts/localisations/translation.js (hunk 2 above)
Comment (Timmaks) [inline app/scripts/localisations/translation.js]: I removed dutch for testing :)

File: app/scripts/views/login.js (hunk 3 above)
Comment (Timmaks) [inline app/scripts/views/login.js]: Ok

File: app/index.html (hunk 4 above)
Comment (Timmaks) [inline app/index.html]: I wanted to add them when the port is OK,  I need to add them indeed to transifex

File: app/scripts/localisations/translation.js (hunk 1 above)
Comment (Timmaks) [inline app/scripts/localisations/translation.js]: Yes, I wanted to remove the whole file but get a error. I can't find where to remove the requirements to this file. That is why it is still there.... I can empty this file for now if you want.

Comment (willyaranda) [inline app/scripts/localisations/translation.js]: Yes, please, do not keep any translation on this file

File: app/index.html (hunk 4 above)
Comment (willyaranda) [inline app/index.html]: Ok, perfect

**Comment by willyaranda** [comment]
@Timmaks just change those small bits, we will merge and start using transifex to translate this app.

thanks!

File: app/index.html (hunk 4 above)
Comment (Timmaks) [inline app/index.html]: we need to decide if we translate the fxosrate files (add them to the properties file) or if we pull the translations from the fxosrate repro. there are only 3 languages there.

We can generate in transifex a separate fxosrate properties file and let our  translators translate them. and do a pull request for fxosrate so they will have more translations in there repro

**Comment by Timmaks** [comment]
@willyaranda done :-)

File: app/index.html (hunk 4 above)
Comment (willyaranda) [inline app/index.html]: I prefer to pull files from their repository (from bower), yeah. And I like the idea of using a PR with our translations to the project, so let's do that way (generate another properties and then PR them to the upstream project)

**Comment by willyaranda** [comment]
Let's merge this. Awesome work @Timmaks!!

**Pull request merged**